### PR TITLE
Changes to analyze ERS1, ERS2 and ASAR

### PR DESCRIPTION
**Pull request by gr4n0t4**
- Changes on driver ESA to read .E1 .E2 and .N1 files
- Changes to geocode to add specific parameters when analyzing this data

### CODE REVIEW

File: pyroSAR/drivers.py [hunk 1]
```diff
@@ -400,7 +404,10 @@ def gdalinfo(self):
             the metadata attributes
         """
         files = self.findfiles(r'(?:\.[NE][12]$|DAT_01\.001$|product\.xml|manifest\.safe$)')
-        
+        # If only one file return the file in array
+        if type(files==str):
```
Comment (johntruckenbrodt) [inline pyroSAR/drivers.py:408]:
```python
if isinstance(files, str):
```

File: pyroSAR/drivers.py [hunk 2]
```diff
@@ -1516,16 +1525,15 @@ def __init__(self, scene):
...
+                       r'(?P<satellite_ID>[EN][12])'
         self.pattern_pid = r'(?P<sat_id>(?:SAR|ASA))_' \
                            r'(?P<image_mode>(?:IM(?:S|P|G|M|_)|AP(?:S|P|G|M|_)|WV(?:I|S|W|_)|WS(?:M|S|_)))_' \
                            r'(?P<processing_level>[012B][CP])'
         
         self.scene = os.path.realpath(scene)
         
-        self.examine()
+        # Only a file to examine
+        self.file = self.scene
```
Comment (johntruckenbrodt) [inline pyroSAR/drivers.py:1536]: Please rewrite it to make sure that `self.file` directly points to the file if the scene is zipped as well as after calling `self.unpack`:
```python
if re.search('.[EN][12]$', self.scene):
    self.file = self.scene
else:
    self.examine()
```

**Review by johntruckenbrodt — CHANGES_REQUESTED** [review]
Hi @gr4n0t4,
thanks for the pull request, much appreciated. I have cleaned up the CI builds, which revealed a failing test in the call of `self.gdalinfo()`. The points addressed in the review should fix it.
Cheers, John

**Comment by gr4n0t4** [comment]
Checked that the changes work, not sure what is the next step

**Comment by johntruckenbrodt** [comment]
@gr4n0t4 you need to make the changes to the code that I suggested and push them to your fork on GitHub. This will update the pull request and rerun the CI tests.

**Comment by gr4n0t4** [comment]
> @gr4n0t4 you need to make the changes to the code that I suggested and push them to your fork on GitHub. This will update the pull request and rerun the CI tests.

I merge the last changes into my fork, and the changes you suggest were there.

**Comment by johntruckenbrodt** [comment]
I don't know what you mean. I have not pushed the changes, only suggested them in the review. Please observe [line 409](https://github.com/SAR-ARD/pyroSAR/blob/7b7af720d8371af6b8b92339fe2c28b891dd06d7/pyroSAR/drivers.py#L409) and change it:
```diff
-        if type(files==str):
+        if isinstance(files, str):
```

**Review by johntruckenbrodt — APPROVED** [review]

**Comment by johntruckenbrodt** [comment]
No we got it. Thanks a lot Ricardo!

**Comment by gr4n0t4** [comment]
Thanks for the support @johntruckenbrodt

**Pull request merged**